govc/sso/user/update.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package user
+
+import (
+	"context"
+	"flag"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/ssoadmin"
+	"github.com/vmware/govmomi/ssoadmin/types"
+)
+
+type update struct {
+	userDetails
+}
+
+func init() {
+	cli.Register("sso.user.update", &update{})
+}
+
+func (cmd *update) Description() string {
+	return `Update SSO users.
+
+Examples:
+  govc sso.user.update -C "$(cat cert.pem)" NAME
+  govc sso.user.update -p password NAME`
+}
+
+func (cmd *update) Run(ctx context.Context, f *flag.FlagSet) error {
+	if f.NArg() != 1 {
+		return flag.ErrHelp
+	}
+	id := f.Arg(0)
+
+	updated := false
+	merge := func(src *string, current string) {
+		if *src == "" {
+			*src = current
+		} else {
```

Comment:
if `src == current`, updated still false, no?

Reply:
I added a comment to the merge func.  And removed the `updated` flag, no need to optimize in this case.